dotcom-rendering/src/web/browser/islands/initHydration.ts

```diff
@@ -49,10 +49,16 @@ export const initHydration = (elements: NodeListOf<Element>): void => {
 							props,
 							element,
 							emotionCache,
-						).then(() => {
-							targetElement.dispatchEvent(
-								new MouseEvent('click'),
-							);
+						).then(({ success }) => {
+							if (success) {
+								setTimeout(() => {
+									targetElement.dispatchEvent(
+										new MouseEvent('click', {
+											bubbles: true,
+										}),
+									);
+								}, 50);
```

Comment:
We need this timeout because even though `doHydration` does wait for the island to be downloaded, we have no way to know when the code inside the island (LightboxButton in this case) has completed. Somewhere in that code we set a listener for the button click and we need to wait for that to happen before firing the synthetic click.

We could use a callback, say `done()`, and use that as a way to know when the island is ready but that would make creating islands more complex. 🤷 maybe the hack works?